### PR TITLE
Add a stub implementation for the Elfie collector

### PR DESCRIPTION
**Pull request by rtaket**
This is templated code taken from the existing *2CatalogCollector classes. The CatalogCollector classes enumerate through all the packages in the NuGet catalog. This CatalogCollector is just stubbed out. i.e. There's currently no Elfie implementation. It just prints trace statements for each package it encounters.

### CODE REVIEW

File: src/Ng/Catalog2Elfie.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    Thread.Sleep(options.Interval * 1000);
+                }
+            }
+        }
+
+        void PrintUsage()
+        {
+            Console.WriteLine();
+            Console.WriteLine("Usage: ng catalog2elfie -source <catalog> -storageBaseAddress <storage-base-address> -storageType file|azure [-storagePath <path>]|[-storageAccountName <azure-acc> -storageKeyValue <azure-key> -storageContainer <azure-container> -storagePath <path>] [-verbose true|false] [-interval <seconds>] [-maxthreads <int>]");
+            Console.WriteLine("Example: ng.exe catalog2elfie -source http://api.nuget.org/v3/catalog0/index.json -storageBaseAddress file:///C:/NuGet -storageType file -storagePath C:\\NuGet -verbose true");
```
Comment (jinujoseph) [inline src/Ng/Catalog2Elfie.cs:67]: whats the difference between storageBaseAddress and storagePath ?

File: src/Ng/Ng.csproj [hunk 2]
```diff
@@ -157,7 +157,10 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Catalog2Dnx.cs" />
+    <Compile Include="Catalog2ElfieOptions.cs" />
```
Comment (jinujoseph) [inline src/Ng/Ng.csproj:160]: So there is no reason down the line we can create a new project with just this new files and what else needed right ?

**Comment by jinujoseph** [comment]
:+1:

File: src/Ng/Catalog2Elfie.cs (hunk 1 above)
Comment (rtaket) [inline src/Ng/Catalog2Elfie.cs:67]: I'll update the usage statement to add descriptions for all the parameters. 

To answer your specific question, storagePath is only used when the storageType==file. It's the directory where the files will be saved. storageBaseAddress makes more sense when the storageType==azure. In this case, storageBaseAddress is the URL to your Azure blob. When storageType==file, storageBaseAddress is really only used to calculate relative paths. For example, if you tell the Storage type to save a stream to file:///C:/NuGet/Idx/Newtonsoft.Json/8.0.2/8.0.2.idx, it removes storageBaseAddress from the URL to determine the relative path (e.g. Idx/Newtonsoft.Json/8.0.2/8.0.2.idx.) Then it saves the file to {storagePath} + {relativePath} = C:\NuGet\Idx\Newtonsoft.Json/8.0.2/8.0.2.idx.

File: src/Ng/Ng.csproj (hunk 2 above)
Comment (rtaket) [inline src/Ng/Ng.csproj:160]: Yes, that would work.

**Pull request merged**